ghcide/src/Development/IDE/Plugin/HLS.hs

```diff
@@ -72,16 +73,17 @@ instance Pretty Log where
           <> pretty method <> ": " <> viaShow exception
 instance Show Log where show = renderString . layoutCompact . pretty
 
-noPluginEnabled :: Recorder (WithPriority Log) -> SMethod m -> [PluginId] -> IO (Either ResponseError c)
+noPluginEnabled :: Recorder (WithPriority Log) -> SMethod m -> [(PluginId, PluginStatus)] -> IO (Either ResponseError c)
 noPluginEnabled recorder m fs' = do
   logWith recorder Warning (LogNoPluginForMethod $ Some m)
   let err = ResponseError (InR ErrorCodes_MethodNotFound) msg Nothing
       msg = pluginNotEnabled m fs'
   return $ Left err
-  where pluginNotEnabled :: SMethod m -> [PluginId] -> Text
+  where pluginNotEnabled :: SMethod m -> [(PluginId, PluginStatus)] -> Text
+        pluginNotEnabled method [] = "No plugin installed for this " <> T.pack (show method) <> " request."
         pluginNotEnabled method availPlugins =
-            "No plugin enabled for " <> T.pack (show method) <> ", potentially available: "
-                <> (T.intercalate ", " $ map (\(PluginId plid) -> plid) availPlugins)
+            "No plugin enabled for this " <> T.pack (show method) <> " request.\n Plugins installed for this method, but not enabled for this request are:\n"
```

Comment:
I think this message is still confusing, and I think it's because of the "enabled" terminology. Something is usually enabled/disabled by an external force, and we have a _different_ use of that (which IMO makes more sense) for plugins that are disabled by the config.

So how about we instead say something like
```
No plugin wanted to handle this request.
Plugins which can handle <method> requests but which declined:
- <plugin 1>: disabled by config
... etc...
```
i.e. go with the "declined" metaphor instead. WDYT?